project/SparkBuild.scala

```diff
@@ -883,6 +884,61 @@ object SparkConnectClient {
   }
 }
 
+object SparkConnectClientCompatibility {
```

Comment:
@hvanhovell Here is a chance to prove me SBT is better? :) 

I tried to run this during assembly on the generated artifact, failed because the jar cannot be locally previously generated.

I also tried to create another shaded jar after normal shading run, but it then removes the normally shaded jar.